ddsrouter_event/include/ddsrouter_event/SignalManager.hpp

```diff
@@ -191,6 +191,9 @@ class SignalManager
 
     //! Variable to store the last unique id registered, so the new one is unique.
     UniqueCallbackId current_last_id_;
+
+    //! Guards access to singleton instance
+    static std::recursive_mutex instance_mutex_;
```

Comment:
NIT
Is this still needed?
With the current changes signal threads should not access `get_instance` anymore, and `static` call is thread safe itself.

Reply:
You are probably right, but I do prefer to keep just in case, as it should have no impact in performance.